dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/adaptor/ESAAdaptor.java

```diff
@@ -204,13 +204,15 @@ private static void install(Product product, ESAAsset featureAsset, Provisioning
                                     featureDefinition, fr, processedEntries, checksumInput, checksumOutput, executableFiles, extattrFiles, checksumsManager);
 
                 }
-            } else if (SubsystemContentType.FEATURE_TYPE == type && !skipFeatureDependencyCheck) {
-                // Look to see if the feature already exists and try to find it in the same place as the current ESA if we don't have it
-                String symName = fr.getSymbolicName();
-                if (!product.containsFeature(symName) && !product.containsFeatureCollection(symName) && !featuresToBeInstalled.contains(symName)) {
-                    throw new InstallException(Messages.PROVISIONER_MESSAGES.getLogMessage("tool.install.missing.feature", featureDefinition.getSymbolicName(), symName));
+            } else if (SubsystemContentType.FEATURE_TYPE == type) {
+                if (!skipFeatureDependencyCheck) {
+                    // Look to see if the feature already exists and try to find it in the same place as the current ESA if we don't have it
+                    String symName = fr.getSymbolicName();
+                    if (!product.containsFeature(symName) && !product.containsFeatureCollection(symName) && !featuresToBeInstalled.contains(symName)) {
+                        throw new InstallException(Messages.PROVISIONER_MESSAGES.getLogMessage("tool.install.missing.feature", featureDefinition.getSymbolicName(), symName));
+                    }
                 }
-            } else if (ibmFeature && !skipFeatureDependencyCheck) {
+            } else if (ibmFeature) {
```

Comment:
I think the skipFeatureDependencyCheck condition for this specific `else if` clause was added to avoid the issue mentioned in my previous comment.  It seems like a coding error.  This clause has nothing to do with dependency checking, since the previous clause handles feature references in the manifest, while this clause handles `file` entries (so it checks whether the file path is specified and then extracts it).

Reply:
And I verified the override as well.

Reply:
Supposedly this clause could say `else if (ibmFeature && SubsystemContentType.FILE_TYPE == type)`, but I'm not sure we should change this since it's not what was in the code before #3719